plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/GenerateDistributionPropertiesTask.kt

```diff
@@ -57,18 +61,44 @@ abstract class GenerateDistributionPropertiesTask : PropertiesFileOutputTask() {
       output: Provider<Directory>,
       taskSuffix: String,
       buildConfiguration: String,
+      sentryOrg: String? = null,
+      sentryProject: String? = null,
+      variant: SentryVariant? = null,
     ): TaskProvider<GenerateDistributionPropertiesTask> {
       return project.tasks.register(
         "generateSentryDistributionProperties$taskSuffix",
         GenerateDistributionPropertiesTask::class.java,
       ) { task ->
         task.output.set(output)
         task.withSentryTelemetry(extension, sentryTelemetryProvider)
-        // TODO we should check if the org and project are available in sentry.properties
-        task.orgSlug.set(extension.org)
-        task.projectSlug.set(extension.projectName)
-        // TODO we should have a separate authToken for Build Distribution
+
+        // Load sentry.properties if available
+        val sentryPropertiesFile = variant?.let { SentryPropertiesFileProvider.getPropertiesFilePath(project, it) }
+        val sentryProperties = sentryPropertiesFile?.let { PropertiesUtil.loadMaybe(File(it)) }
+
+        // Resolve org slug with fallback chain: ext -> extension -> env -> sentry.properties
```

Comment:
Happy to discuss a different priority here.
Note: we're not getting `.sentryclirc` files yet. That can be added later.